src/refract/primitives.es6

```diff
@@ -0,0 +1,322 @@
+/*
+ * A refract element implementation with an extensible type registry.
```

Comment:
Thanks for the feedback!

> We had the concept of this registry where you registered your own types, but is there a way to do this so you don't have to create new type classes? In other words, if an instance of a Refract element knows it's a string, it will always behave like a string no matter name you give it.

You don't strictly have to create your own classes. The way it works now you can do this:

```js
import {StringType, TypeRegistry} from 'primitives';

// I have a few elements I want to register that act like strings
TypeRegistry.elementMap['my-element'] = StringType;
TypeRegistry.elementMap['my-other-element'] = StringType;
TypeRegistry.elementMap.frob = StringType;

// Now I can handle them!
import {fromCompactRefract} from 'primitives';

let elements = fromCompactRefract(['array', {}, {}, [
  ['my-element', {}, {}, 'my-value'],
  ['my-other-element', {}, {}, 'my-other-value'],
  ['frob', {}, {}, 'abc123']
]);

console.log(elements.get(0).content); // => 'my-value'
```

> I don't have any answers at this point, but I feel like getting adoption with this will require an interface that adds just a little complexity to what people are familiar with in the JSON world while providing the extra stuff Refract provides.

I agree. Let me see what I can come up with on this front. It seems like my initial thinking was wrong and we do want to support built-in types easily.

The error PR is pretty interesting as well. I'm not sure how best to handle errors yet, but that sounds like a promising approach.